File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        vector<int> temp;
        for(int i=0;i<nums.size();i++){
            if(nums[i] == val){
                temp.push_back(i);
            }
        }
        for(int i=0;i<temp.size();i++){
            nums.erase(nums.begin() + temp[i] - i);
        }
        return nums.size();
    }
};

File: 0027-remove-element/0027-remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int size = nums.length;
        int left = 0, right = size-1;
        int count = 0;
        while(left <= right){
            if(nums[left] == val){
                count++;
                while(left != right && nums[right] == val){
                    right--;
                    count++;
                }
                nums[left] = nums[right];
                right--;
                left++;
            }else{
                left++;
            }
        }
        return size - count;
    }
}